### ISSUE TITLE
Tips for downgrading this to an older version of tree-sitter

### ISSUE THREAD

**Issue by jonboiser** [stsewd/tree-sitter-comment#3]
Hi! I'm interested in forking this library and rebuilding it for use in an Atom grammar. I tried doing this with the library as is, but because it the language version is 13

https://github.com/stsewd/tree-sitter-comment/blob/674c5cbb948b596c3ed664b510dde76e1f21aee3/src/parser.c#L8

it is incompatible with the current versions of Atom (and Atom might not catch up any time soon).

I'm completely ignorant about tree-sitter parser side of things. Do you think it's a simple matter of just regenerating the C code with an older version of `tree-sitter-cli`, but with the same grammar you have here?

**Comment by stsewd** [comment]
> Do you think it's a simple matter of just regenerating the C code with an older version of tree-sitter-cli, but with the same grammar you have here?

Yeah, you can re-generate the grammar, it should just work. You just need to download the ts cli version that you want and call `tree-sitter-cli generate`, you can also use this commit https://github.com/stsewd/tree-sitter-comment/tree/6b0dbd56284023617c3ea755624e55d0254f813a (that's using 0.17.3). There aren't any differences in the grammar from that commit vs master, so you can use it without problems.

**Comment by jonboiser** [comment]
Thanks! Installing from '6b0dbd5' was a quick fix.

I was wondering if you might know how to work around what seems to be a common limitation with other language grammars: that comments are parsed as single block, which makes [language injection](https://flight-manual.atom.io/hacking-atom/sections/creating-a-grammar/#language-injection) of this grammar hard/impossible.

For example, I'm trying to use the `tree-sitter-vue` parser to update the Atom vue grammar. But it treats the entire comment block as a `comment`, so I can't just feed it into your parser. The `<!--` etc. isn't parsable by `tree-sitter-comment`.

<img width="404" alt="Screen Shot 2021-03-20 at 8 36 13 PM" src="https://user-images.githubusercontent.com/10248067/111893169-efcdd380-89bd-11eb-9b31-78caee89c366.png">

Based on other parsers I've looked at (JS, Python, etc.) That seems to be a common pattern. I think to use your parser, you would need to actually parse the contents of the `comment` block as its own node (e.g. `comment_line`, or `comment_content`), unless there's another way to do this.

I've tried simply removing the parts that are not parsable by `ts-comment`  before passing it on, but I haven't had success with that yet...
